carma_wm/src/WMListenerWorker.cpp

```diff
@@ -483,37 +488,52 @@ void WMListenerWorker::routeCallback(const cav_msgs::RouteConstPtr& route_msg)
 
   }
 
-  recompute_route_flag_ = false;
-  rerouting_flag_ = false;
-
   if (!world_model_->getMap()) { // This check is a bit redundant but still useful from a debugging perspective as the alternative is a segfault
     ROS_ERROR_STREAM("WMListener received a route before a map was available. Dropping route message.");
     return;
   }
 
-  auto path = lanelet::ConstLanelets();
-  for(auto id : route_msg->shortest_path_lanelet_ids)
-  {
-    auto ll = world_model_->getMap()->laneletLayer.get(id);
-    path.push_back(ll);
-  }
-  if(path.empty()) return;
-  auto route_opt = path.size() == 1 ? world_model_->getMapRoutingGraph()->getRoute(path.front(), path.back())
-                               : world_model_->getMapRoutingGraph()->getRouteVia(path.front(), lanelet::ConstLanelets(path.begin() + 1, path.end() - 1), path.back());
-  if(route_opt.is_initialized()) {
-    auto ptr = std::make_shared<lanelet::routing::Route>(std::move(route_opt.get()));
-    world_model_->setRoute(ptr);
+  // If one of the applied queued map updates invalidated the route, then the routing graph must be updated again for the route node
+  if (route_invalidated_by_queued_map_update && route_node_flag_){
+    ROS_DEBUG_STREAM("At least one applied queued map update has invalidated the route. Routing graph will be recomputed.");
+    world_model_->setMap(world_model_->getMutableMap(), current_map_version_, route_invalidated_by_queued_map_update);
+    ROS_DEBUG_STREAM("Finished recomputing the routing graph for the applied queued map update(s)");
+
+    rerouting_flag_ = true; // Set flag to trigger a route update by the route node due to the updated routing graph
+
+    return;
   }
+  else {
+    rerouting_flag_ = false; // Reset flag since no applied queued map updates invalidated the route for the route node
 
-  world_model_->setRouteEndPoint({route_msg->end_point.x,route_msg->end_point.y,route_msg->end_point.z});
+    auto path = lanelet::ConstLanelets();
+    for(auto id : route_msg->shortest_path_lanelet_ids)
+    {
+      auto ll = world_model_->getMap()->laneletLayer.get(id);
+      path.push_back(ll);
+    }
+    if(path.empty()) return;
```

Comment:
Is this a likely case? Should there be an error report here?
I would say that if we hit this case we should also be clearing the world model's route, such that a user node would no longer find a route available.

Reply:
Good point, I don't think it is possible (or at least highly unlikely) for the route node to publish a route message without any lanelet IDs listed in the shortest path. This specific logic was carried over from the initial release/jetta-gillet branch, and I would be fine with removing the `if(path.empty()) return;` line completely.

Reply:
Pushed a commit to remove this check.